tests/Unit/Core/Util/FormChoiceFormatterTest.php

```diff
@@ -0,0 +1,109 @@
+ */
+
+declare(strict_types=1);
+
+namespace Tests\Unit\Core\Util;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Core\Util\FormChoiceFormatter;
+
+class FormChoiceFormatterTest extends TestCase
+{
+    /**
+     * @dataProvider getFormOptionsToFormat
+     *
+     * @param array $rawOptions
+     * @param string $idKey
+     * @param string $nameKey
+     * @param array $expectedFormattedChoices
+     */
+    public function testCleanUrl(array $rawOptions, string $idKey, string $nameKey, array $expectedFormattedChoices): void
+    {
+        $returnedFormattedChoices = FormChoiceFormatter::formatFormChoices($rawOptions, $idKey, $nameKey);
+        $this->assertEquals($expectedFormattedChoices, $returnedFormattedChoices);
+    }
+
+    public function getFormOptionsToFormat(): iterable
+    {
+        yield 'clean absolute url _token' => [
+            [
+                [
+                    'id_manufacturer' => 1,
+                    'name' => 'Krystian and son development',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 2,
+                    'name' => 'Preston Manufacturing',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 3,
+                    'name' => 'Trendo',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 4,
+                    'name' => 'Hiba Manufacturing',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 5,
+                    'name' => 'Krystian and son development',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 6,
+                    'name' => 'Daniel Manufacturing',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 7,
+                    'name' => 'Krystian and son development',
+                    'someproperty' => 'somevalue',
+                ],
+                [
+                    'id_manufacturer' => 8,
+                    'name' => 'Hiba Manufacturing',
+                    'someproperty' => 'somevalue',
+                ],
+            ],
+            'id_manufacturer',
+            'name',
+            [
+                'Daniel Manufacturing' => 6,
+                'Hiba Manufacturing (4)' => 4,
+                'Hiba Manufacturing (8)' => 8,
+                'Krystian and son development (1)' => 1,
+                'Krystian and son development (5)' => 5,
+                'Krystian and son development (7)' => 7,
+                'Preston Manufacturing' => 2,
+                'Trendo' => 3,
+            ],
+        ];
```

Comment:
You should also add an use case where there is no duplicates, to validate that in this case the names are not modified at all (which is 90% of the time probably)